### PR TITLE
Fs 2432 start page

### PR DESCRIPTION
**Pull request by srh-sloan**
https://digital.dclg.gov.uk/jira/browse/FS-2432
https://digital.dclg.gov.uk/jira/browse/FS-2183
https://app.mural.co/t/mhclg2837/m/mhclg2837/1675177182644/578e9019ed4a6da033b5de2183289ca7298e5f12?sender=u94fce4b3d87aef5380039386

### Change description
These changes are to make the fund start page work for any fund, based on URL path. The existing index page (`/`) now redirects to `/cof/r2w3` as COF R2W3 is still live and uses that as it's start page. However all funds that exist in the fund store will now be accessible at `/fund_short_name/round_short_name` eg. `/cof/r2w2` or `/COF/R2W3`.

If you try and access this using just a fund short code (eg. `/cof/`) you will get redirected to the most recent round in that fund.

Added a new template `fund_start_page.html` which parameterises all the text on the start page and these values are now pulled from the fund store.

- [x] Unit tests and other appropriate tests added or updated
- [x] README and other documentation has been updated / added (if needed)
- [x] Commit messages are meaningful and follow good commit message guidelines (e.g. "FS-XXXX: Add margin to nav items preventing overlapping of logo")

### How to test
Run locally with the docker runner. Navigate to the following URLs
- http://localhost:3008/ - should render as currently
- http://localhost:3008/cof/r2w3 - should render as above
- http://localhost:3008/cof/r2w2 - should display Round 2 Window 2 as closed
- http://localhost:3008/cof - redirects to r2w3 page above (As that's the most recently open window)
- http://localhost:3008/made_up_fund - 404

### Screenshots of UI changes (if applicable)
No visible changes to users

### CODE REVIEW

File: app/default/account_routes.py [hunk 1]
```diff
@@ -92,6 +83,12 @@ def build_application_data_for_display(applications: list[ApplicationSummary]):
 @login_required
 def dashboard():
     account_id = g.account_id
+
+    # fund_short_name = request.args.get("fund")
+    # round_short_name = request.args.get("round")
```
Comment (gio-karunakaran) [inline app/default/account_routes.py:88]: Do we want to keep these commented lines?

File: app/default/routes.py [hunk 2]
```diff
@@ -1,46 +1,104 @@
...
     current_datetime_after_given_iso_string,
 )
 
-
 default_bp = Blueprint("routes", __name__, template_folder="templates")
 
 
+# TODO Do not change this until after COF R2 W3 closes as this
+# provides the current landing page!
 @default_bp.route("/")
```
Comment (gidsg) [inline app/default/routes.py:22]: Can we make this a `302` redirect to the COF R2 W3 specific page? This way we could start using that without breaking any existing links?

File: tests/conftest.py [hunk 3]
```diff
@@ -102,3 +103,17 @@ def mock_get_application(mocker):
...
+    recorded = []
+
+    def record(sender, template, context, **extra):
+        recorded.append((template, context))
+
+    template_rendered.connect(record, app)
+    try:
+        yield recorded
+    finally:
+        template_rendered.disconnect(record, app)
```
Comment (gio-karunakaran) [inline tests/conftest.py:119]: very funky! love it 🙌🏾

**Review by gio-karunakaran — DISMISSED** [review]
LGTM 👍🏾

File: app/default/account_routes.py (hunk 1 above)
Comment (srh-sloan) [reply app/default/account_routes.py:88]: Yeah, they are kind of a reminder to me for the next part in all this with the dashboard if that's ok

File: app/default/routes.py (hunk 2 above)
Comment (srh-sloan) [reply app/default/routes.py:22]: Oh yeah, good shout 👍

**Pull request merged**